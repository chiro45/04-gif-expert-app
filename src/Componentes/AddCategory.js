import React, {useState} from 'react'
import PropTypes from 'prop-types';
export default function AddCategory({setCategories}) {
    //declaramos el useState
    const [inputValue, setinputValue] = useState("");
    //el evento debe introducir el target.value
    const handleInputChange = (e)=>{
        setinputValue(e.target.value)
    }
    //
    const handleSubmit = (e)=>{
        //le decimos que no realice el evento default
        e.preventDefault(e)
        //imprime en consola
        console.log('handle submit', inputValue)
        //validamos que lo introducido sea mayor a 2
        if(inputValue.trim().length > 2){
        
         setCategories( categories => [inputValue, ...categories,])
         setinputValue('')

        }
        //console.log('submit hecho')
    }
    return (    
        //retorna el formulario con el evento y el cambio 
        <form onSubmit={handleSubmit}>
        <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}
        />
        </form>   
    )
}
//define las propiedades del componente
AddCategory.propTypes ={
    setCategories: PropTypes.func.isRequired
}