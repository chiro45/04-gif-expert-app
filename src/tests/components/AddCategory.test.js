import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from "enzyme"
import AddCategory from "../../Componentes/AddCategory"

describe('Pruebas en el componente <AddCategory/>', () => {

    const setCategories = jest.fn();
    let wrapper =  shallow(<AddCategory setCategories={setCategories}/>);

    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>)
    });

    test('debe mostrase correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    test('debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = "hello world"

        input.simulate('change',{ target : { value } });
        expect(wrapper.find('p').text().trim()).toBe(value)

    })
    test('no debe de psostear la informacion onSubmit', () => {
       wrapper.find('form').simulate('submit',{preventDefault(){}});
        expect(setCategories).not.toHaveBeenCalled();
    })
    test('debe de llamar el setCategories y liimpiar la caja de texto', () => {
        
        const value = "hello world";
        //1.Simular el inpuChange
        wrapper.find('input').simulate('change',{ target : { value } });
        //2. Simular el submit
        wrapper.find('form').simulate('submit',{ preventDefault(){}} );
        //3.setCategories se debe haber llamado una vez
        expect(setCategories).toHaveBeenCalled();
        //4.el valor del input debe estar ""
        expect(wrapper.find('input').prop('value')).toBe('')

    })
})
