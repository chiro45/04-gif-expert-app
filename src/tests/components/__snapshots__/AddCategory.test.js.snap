// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AddCategory/> debe mostrase correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
